src/easylink/pipeline_graph.py

```diff
@@ -87,26 +207,84 @@ def merge_combined_implementations(self, config):
+        ----------
+        nodes
+            Sorted names of the nodes to validate.
+        metadata_steps
+            Sorted names of the steps that the nodes are intended to implement.
+
+        Raises
+        ------
+        ValueError
+            If there is a mismatch between the nodes and the steps they intend
+            to implement or if the nodes are no longer topologically consistent.
+        """
+        # HACK: We cannot just call self.subgraph(nodes) because networkx will
+        # try and instantiate another PipelineGraph which requires a Config object
+        # to be passed to the constructor.
+        subgraph = ImplementationGraph(self).subgraph(nodes)
+
+        # Relabel nodes by schema step
+        mapping = {
+            node: data["implementation"].schema_step
+            for node, data in subgraph.nodes(data=True)
+        }
+        if not set(mapping.values()) == set(metadata_steps):
+            # NOTE: It's possible that we've combined nodes in such a way that removed
+            # an edge from the graph and so nx is unable to reliably sort the subgraph.
+            full_pipeline_sorted_nodes = list(nx.topological_sort(self))
+            sorted_nodes = [node for node in full_pipeline_sorted_nodes if node in mapping]
+            raise ValueError(
+                f"Pipeline configuration nodes {[mapping[node] for node in sorted_nodes]} "
+                f"do not match metadata steps {metadata_steps}."
+            )
+        subgraph = nx.relabel_nodes(subgraph, mapping)
+        # Check for topological inconsistency, i.e. if there is a path from a later
+        # node to an earlier node.
+        for predecessor in range(len(metadata_steps)):
+            for successor in range(predecessor + 1, len(metadata_steps)):
+                if nx.has_path(
+                    subgraph, metadata_steps[successor], metadata_steps[predecessor]
+                ):
+                    raise ValueError(
+                        f"Pipeline configuration nodes {list(nx.topological_sort(subgraph))} "
+                        "are not topologically consistent with the intended implementations "
+                        f"for {list(metadata_steps)}:\n"
+                        f"There is a path from successor {metadata_steps[successor]} "
+                        f"to predecessor {metadata_steps[predecessor]}."
+                    )
+
     def _get_combined_slots_and_edges(
         self, combined_implementation: str, nodes_to_merge: list[str]
     ) -> tuple[set[InputSlot], set[OutputSlot], set[EdgeParams]]:
-        """Get all the input slots, output slots, and edges needed to construct the
-        combined implementation
+        """Gets all edge information required for the combined implementation.
 
         Parameters
         ----------
         combined_implementation
-            Name of the combined implementation.
+            The name of the combined implementation.
         nodes_to_merge
```

Comment:
```suggestion
        nodes_to_combine
```
?

Reply:
Actually went with "merge" instead of "combine"